spec/unit/hanami/web/rack_logger_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "hanami/web/rack_logger"
+require "hanami/logger"
+require "stringio"
+require "rack/mock"
+
+RSpec.describe Hanami::Web::RackLogger do
+  subject { described_class.new(logger) }
+  let(:logger) { Hanami::Logger.new(application_name, stream: stream, level: Hanami::Logger::DEBUG, filter: filters) }
+  let(:stream) { StringIO.new }
+  let(:filters) { ["user.password"] }
+  let(:application_name) { "my_app" }
+
+  describe "#initialize" do
+    it "returns an instance of #{described_class}" do
+      expect(subject).to be_kind_of(described_class)
+    end
+  end
+
+  describe "#log_request" do
+    it "logs current request" do
+      path = "/users"
+      ip = "127.0.0.1"
+      status = 200
+      time = 0.0001
+      content_length = 23
+      verb = "POST"
+
+      env = Rack::MockRequest.env_for(path, method: verb)
+      env["Content-Length"] = content_length
+      env["REMOTE_ADDR"] = ip
+
+      params = {"user" => {"password" => "secret"}}
+      env["router.params"] = params
+
+      subject.log_request(env, status, time)
+
+      stream.rewind
+      actual = stream.read
+
+      expect(actual).to include(%([#{application_name}] [INFO] [#{time}] #{verb} #{status} #{ip} #{path} #{content_length} {"user"=>{"password"=>"[FILTERED]"}}))
```

Comment:
```suggestion
      expect(actual).to include(%([my_app] [INFO] [0.0001] POST 200 127.0.0.1 /users 23 {"user"=>{"password"=>"[FILTERED]"}}))
```
Thoughts on this? In my experience it's a little more robust, since renaming a variable above could make one of these into `nil`, and `#{nil}` is an empty string and we wouldn't know